moved sharktank-ci and regression-test-ci to the ossci cluster

Eliasj42 opened the pull request:
moved sharktank-ci and regression-test-ci to the ossci cluster

benvanik committed:
Cleaning up typos and code style.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Adding ElideAsyncTransfersPass to try to remove redundant transfers.
These transfers can arise during stream conversion if there was
ambiguity in the source IR.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Adding CloneToConsumersPass to run prior to stream conversion.
This performs affinity analysis and clones any clonable producer that is
used by multiple affinities per-user. It's effectively the inverse of a
CSE step and is intended to undo CSE that created the affinity ambiguity.

The primary case where this arises today is CSE collapsing a splat or
splat-like dispatch that is consumed by transfers or dispatches on
different affinities.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Fixing PropagateClonableOps canonicalizer for multi-result/types.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Temporarily disabling the ElideAsyncTransfers pass.
There's larger issues to fix in resource usage analysis and lifetime
assignment.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Adding utility print to resource usage analysis.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
refreshed PR to hopefully address unrelated failures

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
refreshed PR to hopefully address unrelated failures

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Groverkss committed:
[ExternalInterfaces] Fix LinalgExtFusionInterface registeration (#20244)

addExtension adds a dialect dependency on the dialect passed as an
argument. If LinalgExtDialect is never registered, we would never
register LinalgExtFusionInterface for linalg ops.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

bjacob committed:
Fix floating point type conversions when the result is in the denormal range (#20242)

The condition for generating a denormal (which we just flush to zero)
was incorrect. The new condition is very simple, checks that the biased
converted exponent would be <= 0 which is the definition of a denormal.
The added testcases show what is being fixed here. Prior to the fix,
these cases converted to some large values as the exponent wrapped
around!

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
changed test tolerations

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
changed test tolerations

Signed-off-by: Elias Joseph <eljoseph@amd.com>

IanWood1 committed:
[docs] Update Optimization Options (#20216)

Added `iree-opt-level` to docs and add tip pointing to the "Optimization
Options" page.

---------

Signed-off-by: Ian Wood <ianwood2024@u.northwestern.edu>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
changed test tolerations

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
changed test tolerations

Signed-off-by: Elias Joseph <eljoseph@amd.com>

bjacob committed:
Fix libdevice `iree_f2h_ieee` conversion (#20248)

This replaces the implementation with a specialized copy of the code we
have in base/internal/math.h.

This also adds an extensive test, so that we can feel better about this
code which isn't shared with anything else (being libdevice) and is
relatively little used, being used only with workloads involving the
CPU-hostile f16 type, and only on CPU targets lacking native f16-f32
conversion instructions, which is generally only `generic` CPU targets
as contemporary CPUs tend to have native instructions for these f16-f32
(F16C extension on x86) even though they lack native f16 *arithmetic*
beyond these conversions.

Fixes #20163.

---------

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

ziereis committed:
Fixing missing Metal HAL allocator parent device reference. (#20254)

Signed-off-by: Thomas Ziereis <ziereis@roofline.ai>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Removing use if the `iree_input` dialect from linalgext. (#20255)

Signed-off-by: Elias Joseph <eljoseph@amd.com>

bjacob committed:
[Runtime] Preserve denormals in floating point conversions (#20252)

These helper functions are used notably for printing and parsing, so
flushing denormals to zero there was not a good idea, even if actual
codegen would flush to zero. For example, flusing to zero when printing
could hide denormals, and parsing to zero could prevent reproducing
issues with denormals.

A couple of unrelated changes are lumped in this PR: dropping some
redundant branches to handle `nan_as_neg_zero` (which is used in FP8
types that have FN in their name). I just spotted that and it's
redundant because it's already handled at the end of the function, and
the tests still passing confirm that.

As rightly called out by @krzysz00 in
https://github.com/iree-org/iree/pull/20242#pullrequestreview-2686197346.

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Deleting `iree_input` dialect. (#20256)

The dialect is no longer used and we encourage projects to lower
directly to the appropriate HAL and flow ops. The input dialect predates
MLIR's versioning mechanism and we weren't versioning iree_input
manually anyway so it was just adding an extra layer of goo - if in the
future we want to version the input ops in the internal dialects we can
do so using MLIR's versioning mechanism.

The IREEImportPublic pass is still used for func->util conversion today
and will be removed in a follow-up PR.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

bangtianliu committed:
Integrates/llvm 20250314 llvm/llvm-project@e45090e (#20246)

bump to
https://github.com/llvm/llvm-project/commit/e45090e5f0bf7743fe0b00d510a903a659354ce1

cherry-pick revert commit
https://github.com/iree-org/llvm-project/commit/d5d87f966fd9e0ccf0d1e48b44fe86e35c05f4c8
to https://github.com/llvm/llvm-project/pull/129850

cherry-pick revert commit
https://github.com/iree-org/llvm-project/commit/3e159324ada1aa14444caf2e1a3ef8b0c62109a1
to https://github.com/llvm/llvm-project/pull/130487

add commit
https://github.com/iree-org/llvm-project/commit/a1fc066d52e7fbc783c1cef21cb13752b4db34fa
to fix msvc debug build.

---------

Signed-off-by: Bangtian Liu <liubangtian@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Hardcode84 committed:
[python] Register `vector.transform` ops (#20264)

Also, add a simple smoke test for upstream dialects import.

---------

Signed-off-by: Ivan Butygin <ivan.butygin@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd committed:
[ci] Clean up runner-env and runner-group tags. (#20175)

These were used by the prior GCP runner cluster
(https://github.com/iree-org/iree/issues/18238,
https://github.com/iree-org/iree/issues/17893) but new self-hosted
runners do not currently use them.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

hanhanW committed:
Introduce Flow/Stream tensor encode op and the conversion. (#20251)

The revision introduces `flow.tensor.encode` op and
`stream.tensor.encode` op, and the conversion from Flow to Stream.

It is a step towards https://github.com/iree-org/iree/issues/20187

---------

Signed-off-by: hanhanW <hanhan0912@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

kuhar committed:
[Codegen] Do not pad very small dims (#20272)

It may not be worth to padd reduction dimentions that do not even span
all cache sets.

Ported from:
https://github.com/iree-org/iree/commits/shared/pad-sdxl-permissive/

Signed-off-by: Jakub Kuderski <jakub@nod-labs.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd committed:
Clarify error message for missing module dependencies. (#20036)

For pipelines like
https://github.com/iree-org/iree/blob/main/experimental/benchmarks/sdxl/sdxl_pipeline_bench_f16.mlir,
the order that each .vmfb module is passed to tooling matters.

Running for example
```bash
iree-run-module \
  --module=sdxl_full_pipeline_fp16_rocm.vmfb \
  --module=sdxl_clip_vmfbs/model.rocm_gfx942.vmfb \
  --module=sdxl_unet_fp16_vmfbs/model.rocm_gfx942.vmfb \
  --module=sdxl_vae_vmfbs/model.rocm_gfx942.vmfb \
  ...
```

instead of

```diff
iree-run-module \
- --module=sdxl_full_pipeline_fp16_rocm.vmfb \
  --module=sdxl_clip_vmfbs/model.rocm_gfx942.vmfb \
  --module=sdxl_unet_fp16_vmfbs/model.rocm_gfx942.vmfb \
  --module=sdxl_vae_vmfbs/model.rocm_gfx942.vmfb \
+ --module=sdxl_full_pipeline_fp16_rocm.vmfb \
  ...
```

will hit this error message.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
As on all PRs, please add a descriptive title to the PR that will work as a git commit message.

https://iree.dev/developers/general/contributing/#merging-approved-changes

> The squashed commit summary should match the PR title and the commit description should match the PR body (this is the default behavior). Accordingly, please write these as you would a helpful commit message.

Something like "Migrate pkgci_regression_test mi300 jobs to ossci cluster"

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
```diff
@@ -48,12 +48,12 @@ jobs:
             backend: rocm
             sku: mi300
-            runs-on: nodai-amdgpu-mi300-x86-64
+            runs-on: linux-mi300-1gpu-ossci-iree-org
```
Looks like these runners aren't getting cache hits: https://github.com/iree-org/iree/actions/runs/13866799942/job/38901067529?pr=20260#step:6:221

```
Fri, 14 Mar 2025 23:26:00 GMT INFO     ireers_tools.artifacts:artifacts.py:198   Downloading 'real_weights.irpa' (4.78 GiB) to '/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16/real_weights.irpa'
Fri, 14 Mar 2025 23:33:40 GMT **************************************************************
Fri, 14 Mar 2025 23:33:40 GMT Exec: iree-run-module --device=hip --module=/home/runner/_work/iree/iree/model_output_artifacts/sdxl_unet_fp16_960_1024_vmfbs/stable_diffusion_xl_base_1_0_bs1_64_960x1024_fp16_unet.rocm_gfx942.vmfb --function=run_forward --parameters=model=/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16/real_weights.irpa --module=/home/runner/_work/iree/iree/model_output_artifacts/sdxl_unet_fp16_960_1024_vmfbs/stable_diffusion_xl_base_1_0_bs1_64_960x1024_fp16_unet.rocm_gfx942.vmfb --expected_f16_threshold=0.705f --input=@/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16_960_1024/input1.npy --input=@/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16_960_1024/input2.npy --input=@/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16_960_1024/input3.npy --input=@/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16_960_1024/input4.npy --input=@/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16_960_1024/input5.npy --input=@/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16_960_1024/input6.npy --expected_output=@/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16_960_1024/golden_out.npy
```
The uncached file is at `--parameters=model=/home/runner/_work/iree/iree/artifacts/sdxl_unet_fp16/real_weights.irpa`

Compare with https://github.com/iree-org/iree/actions/runs/13864227149/job/38901139153#step:6:221
```
Fri, 14 Mar 2025 19:58:32 GMT INFO     ireers_tools.artifacts:artifacts.py:191   Skipping 'real_weights.irpa' download (4.78 GiB) - local MD5 hash matches
Fri, 14 Mar 2025 19:58:34 GMT **************************************************************
Fri, 14 Mar 2025 19:58:34 GMT Exec: iree-run-module --device=hip --module=/home/esaimana/actions-runner-4/_work/iree/iree/model_output_artifacts/sdxl_unet_fp16_960_1024_vmfbs/stable_diffusion_xl_base_1_0_bs1_64_960x1024_fp16_unet.rocm_gfx942.vmfb --function=run_forward --parameters=model=/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16/real_weights.irpa --module=/home/esaimana/actions-runner-4/_work/iree/iree/model_output_artifacts/sdxl_unet_fp16_960_1024_vmfbs/stable_diffusion_xl_base_1_0_bs1_64_960x1024_fp16_unet.rocm_gfx942.vmfb --expected_f16_threshold=0.705f --input=@/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16_960_1024/input1.npy --input=@/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16_960_1024/input2.npy --input=@/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16_960_1024/input3.npy --input=@/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16_960_1024/input4.npy --input=@/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16_960_1024/input5.npy --input=@/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16_960_1024/input6.npy --expected_output=@/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16_960_1024/golden_out.npy
```
The cached file is at `/home/esaimana/iree_tests_cache/artifacts/sdxl_unet_fp16/real_weights.irpa`

IIRC that's from the `IREE_TEST_FILES` environment variable set on the runner. These runners do have some persistent storage somewhere. Let's find a way to make that cache mechanism compatible with these runners.

cc @geomin12 who was also refactoring these tests over in https://github.com/iree-org/iree-test-suites/tree/main/sharktank_models

ScottTodd replied:
You could try this pattern. I think it would cover all cases we currently care about.

```yml
      # Set variables in some instances of the matrix
      matrix:
        include:
          ...
          - name: amdgpu_rocm_mi300_gfx942
            rocm-chip: gfx942
            backend: rocm
            sku: mi300
            runs-on: nodai-amdgpu-mi300-x86-64
            runs-on: linux-mi300-1gpu-ossci-iree-org
            env-overrides:
              IREE_TEST_FILES: /shark-cache/data/iree-regression-cache

    # Forward environment variables from the matrix context into the env,
    # taking care to not overwrite if the variable is already set.
    - name: Forward matrix environment variables (version 1)
      run: |
        if [ -z "${IREE_TEST_FILES}" ]; then 
          echo "IREE_TEST_FILES=${{ matrix.env-overrides.IREE_TEST_FILES}}" >> $GITHUB_ENV
        fi

    # ... or this could also only set if the override is set. Either approach should work:
    - name: Forward matrix environment variables (version 2)
      run: |
        if [ -n "${{ matrix.env-overrides.IREE_TEST_FILES }}" ]; then 
          echo "IREE_TEST_FILES=${{ matrix.env-overrides.IREE_TEST_FILES}}" >> $GITHUB_ENV
        fi
```

Writing into `GITHUB_ENV` sets the environment for all future steps in the job, no need for extra code in each following job (either in `run` or in `env`) as you have it right now

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable

We used that env var forwarding in https://github.com/nod-ai/shark-ai/commit/0954febc109ea29eed1b0f4f29abc8fc2c827930 recently too.

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
```diff
@@ -115,7 +115,7 @@ jobs:
             --capture=no \
             --log-cli-level=info \
-            --timeout=600 \
+            --timeout=1200 \
```
This timeout shouldn't need to be changed. Cache misses are likely most of the time increase.

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
```diff
@@ -48,12 +48,12 @@ jobs:
             backend: rocm
             sku: mi308
-            runs-on: nodai-amdgpu-mi308-x86-64
+            runs-on: linux-mi300-1gpu-ossci-iree-org
```
This should still be a mi308 machine. Are there mi308 machines in the ossci cluster?

Eliasj42 replied:
The machine I'm working on only has mi300.  I'm not sure if we have a mi308 in the fleet

Groverkss committed:
[CPU] Improve Fusions and KernelConfig heuristics for Attention (#20245)

- Use newer TileAndFuse passes for LinalgExt TileAndVectorize pipeline
- Restrict QK and PV vector tile sizes based on register space available
- Restrict vector tile sizes based on the max allowed vector tile size
verification check. This requires us to move the flat for setting max
vector size to LLVMCPU/Utils.

This PR makes sure we never generate tile sizes that are too big to pass
large vector verification check, i.e. attention will never fail to
compile on CPU. The performance can be improved by selecting better
heuristics, but that is a problem for another day.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd commented on .github/workflows/pkgci_test_sharktank.yml:
```diff
@@ -32,8 +32,8 @@ jobs:
-            gpu: gfx1100
-            runs-on: nodai-amdgpu-w7900-x86-64
+            gpu: gfx942
+            runs-on: linux-mi300-2gpu-ossci-iree-org
```
How about keeping the existing w7900 job and adding a new mi300 job instead of swapping here? You could even add the new job in a separate PR.

We should be testing on as many configurations as possible:

* CPU
* 1 consumer GPU
* 2 consumer GPUs
* 1 datacenter GPU
* 2 datacenter GPUs

Eliasj42 replied:
Ok is w7900 consumer and mi300 datacenter right?

ScottTodd replied:
Close enough in this case. AMD PRO W7900 is targeted at professionals and not for gaming, ideally we'd also test on gaming cards like AMD RX 9070. Both are RDNA cards while mi300 is a datacenter CDNA card.

jtuyls committed:
[DT] Add bcast_map to user_indexing_maps in EncodingAttr (#20210)

This PR moves `bcast_map` into the list of user indexing maps by using a
mix of `ArrayAttr` and `AffineMapAttr`: `[#map1, [#map2, #bcast_map],
#map3]`.

This PR contains no new functionality, so no new tests are added.

Resolves: #20085

Signed-off-by: Jorn Tuyls <jorn.tuyls@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd committed:
Limit CTEST_PARALLEL_LEVEL in linux_x64_clang job. (#20259)

Checking if this helps with flaky timeouts like
https://github.com/iree-org/iree/actions/runs/13865892809/job/38804796013.

This doesn't add much to total run time.

* Random sample took 32 seconds to run tests using max parallelism:
https://github.com/iree-org/iree/actions/runs/13861707751/job/38791587212#step:7:12.
* This PR took 39 seconds to run tests using 32:
https://github.com/iree-org/iree/actions/runs/13866660657/job/38807040519?pr=20259#step:7:13

ci-exactly: linux_x64_clang,
Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd committed:
Introduce a new process for requesting iree-org membership. (#20174)

We had been using a Google Form to process requests to join the
https://github.com/iree-org organization and receive triage or commit
access to repositories. That form was created a while ago under an
account that is less active in the community now and the form is no
longer accepting responses.

The new process I'm proposing has users send a message to a group (email
list) within the LF AI & Data Foundation.

* The group name is `iree-github-requests` (open to other suggestions)
* The group is private, so only project maintainers (technically the set
of iree-org "owners" on github who can send invites) will be able to
read requests
* Requests are encouraged to use a template and the website uses a
`mailto` link with the template provided
* Project maintainers can discuss requests privately on the list and/or
reply to the original sender when an invite has been sent
* The prior form had no mechanism for this, so maintainers could do
duplicate work to check and grant access

We can also compare our process with that for other projects like
https://llvm.org/docs/DeveloperPolicy.html#obtaining-commit-access.

Website preview:
https://scotttodd.github.io/iree/developers/general/contributing/#obtaining-commit-access

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
addressed comments about including tests on other hardware

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
addressed comments about including tests on other hardware

Signed-off-by: Elias Joseph <eljoseph@amd.com>

bjacob committed:
[ROCm] Do rewrite `math.fpowi` (#20281)

When in #20222 we dropped all math rewrites on ROCm, we lost an
important rewrite of `math.fpowi` to multiplications. This caused the
regression flagged by @AmosLewis in
https://github.com/iree-org/iree/pull/20222#issuecomment-2727793538.

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Hardcode84 committed:
[python] Remove linalg import from the test (#20273)

linalg dialect import pulls yaml dependency, which is not present on
some CI runners.

---------

Signed-off-by: Ivan Butygin <ivan.butygin@gmail.com>
Co-authored-by: Scott Todd <scott.todd0@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
added IREE_TEST_FILES env variable

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
added IREE_TEST_FILES env variable

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Max191 committed:
[DataTiling][GPU] Add layout resolver option to ROCMTarget (#20250)

Add a new flag option to ROCMTarget called
`iree-hip-encoding-layout-resolver`. The option selects between the
different layout resolvers for encoding attributes:
 - `none`: Select the default layout resolver (materialize to nop)
- `pad`: Select the GPUPadLayoutAttr resolver (materialize into padding)
- `data-tiling`: Select the GPUEncodingLayoutAttr resolver (data tiling
materialization)

The `iree-hip-enable-experimental-pad-layout` and
`iree-llvmgpu-experimental-data-tiling` are removed because they are
folded into the new option.

Currently, we run both the `MaterializeDeviceEncodingPass` and the
`MaterializeEncodingIntoPaddingPass` because the padding layout
materialization cannot be handled by the MaterializeDeviceEncodingPass.
Supporting pad materialization through the MaterializeDeviceEncodingPass
is left as TODO.

This PR is a step towards https://github.com/iree-org/iree/issues/20202

---------

Signed-off-by: Max Dawkins <max.dawkins@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Max191 committed:
[Stream] Put print behind LLVM_DEBUG (#20282)

Signed-off-by: Max Dawkins <max.dawkins@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

hanhanW committed:
[Stream][NFC] Delete iree-stream-experimental-specialize-encodings flag. (#20279)

Signed-off-by: hanhanW <hanhan0912@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

hanhanW committed:
[Stream] Add encoding specialization support for stream.tensor.encode op (#20278)

Signed-off-by: hanhanW <hanhan0912@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

hanhanW committed:
[Encoding][NFC] Add builders for PadEncodingLayoutAttr. (#20285)

It switches the existing code to use the new builders.

Signed-off-by: hanhanW <hanhan0912@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

kuhar committed:
[Codegen] Do not pad matvec-like problems (#20284)

We do not expect benefits on these since these typically go through
subgroup-reduction-style codegen pipeline with a different memory access
pattern.

Signed-off-by: Jakub Kuderski <jakub@nod-labs.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

MaheshRavishankar committed:
[Preprocessing] Form single dispatch programs. (#20265)

The `MakeSingleDispatchForFunction` pass was an initial attempt to move
the entire program into a single dispatch. It was fairly basic and didnt
account for ABI etc. This PR changes the logic of the pass to find the
core computation of the function using slices and ignores ABI related
operations (and some additional ones).

The pass would previously handle multiple basic-blocks. We should be
able to handle them with the new approach, but needs some iteration on
how the upstream slicing utilities can be used in that context. Dropping
support for that for now, and will be revisited later. This pass is not
really used anywhere, so this "support" is not very meaningful.

---------

Signed-off-by: MaheshRavishankar <mahesh.ravishankar@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

MaheshRavishankar committed:
[Preprocessing] Clarify pre-processing preferences. (#20266)

There are many ways today of running pre-processing passes. Clarify the
preference of modes, and ensure only one path is taken. Current paths
and preferences in decreasing order are

1. Command line options a. Textual pass pipeline description b.
Transform dialect spec file c. PDL spec file

2. (default) Plugin defined pipeline extensions.

This change should be mostly NFC.

---------

Signed-off-by: MaheshRavishankar <mahesh.ravishankar@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
giving permissions

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
giving permissions

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
print debugging

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
print debugging

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
print debugging

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
print debugging

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
temporarilt increased timeout to increase cache

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
temporarilt increased timeout to increase cache

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
reverted timeouts

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
reverted timeouts

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved env definition

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Groverkss committed:
[DispatchCreation] Allow expand shape propagation through attention reduction dimensions (#20221)

Co-authored-by: Ian Wood <ianwood2024@u.northwestern.edu>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Max191 committed:
[ConstEval] Don't create ops with partially uninitialized ConstValueInfo (#20228)

HoistIntoGlobals asserted that any op with ConstValueInfo on one
OpResult should have ConstValueInfo on all OpResults. This is was
necessarily always true in some cases. For example:

```
util.func public @main(%arg0: i32, %arg1: i32) -> (i32, i32) {
  %cst = arith.constant 1 : i32
  %barrier0:2 = util.optimization_barrier %arg0, %arg1 : i32, i32
  %barrier1:2 = util.optimization_barrier %cst, %barrier0#0 : i32, i32
  util.return %barrier1#0, %barrier1#1 : i32, i32
}
```

In this case, we expand the analysis frontier to `%barrier1`, and we
assign producers to each of its OpResults (`%cst` and `%barrier0#0`).
However, since the op is known non-constant, we stop expanding the
frontier for efficiency. This means that `%barrier0` will not be
expanded, and its second OpResult will never be added to the analysis
graph.

We don't want to sacrifice efficiency in cases where ops are known
non-constant. We also need to attach producers to OpResults, even when
its owner is known non-constant, because the information may be needed
to find legal escapes of the producer.

This PR initializes ConstValueInfo on all results of producer operations
when expending the analysis frontier, instead of just the consumed
result. Initializing the ConstValueInfo does not affect the analysis,
and protects against any op having a mix of initialized and
uninitialized ConstValueInfo on its results.

fixes https://github.com/iree-org/iree/issues/20206

---------

Signed-off-by: Max Dawkins <max.dawkins@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Max191 committed:
[ConstEval] Expand to parents of nested immediate root uses (#20214)

This was done during worklist processing in
https://github.com/iree-org/iree/pull/19062, but not for the immediate
consumers of constant roots.

Also move the logic from worklist processing and root creation into a
shared private function. Git blame shows that this logic has been
mistakenly updated in only one of the two places twice now, so this will
hopefully prevent this from happening in the future.

---------

Signed-off-by: Max Dawkins <max.dawkins@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

kuhar committed:
[ROCm] Fix WGP counts for Radeon cards (#20291)

These are meant to be used in distribution/scheduling heuristics like
workgroup reordering and should use the level of granularity used for
dispatching workgroups.

This also aligns with the SIMDs-per-WGP constant which was (correctly)
set to 4 already.

Signed-off-by: Jakub Kuderski <jakub@nod-labs.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

MaheshRavishankar committed:
[Preprocessing] Apply a pass-pipeline based on attributes on func-like ops. (#20287)

This PR adds a new mechanism for running some preprocessing passes on
func-like ops. This is done by use of a new attribute
`preprocessing-passpipeline` which has the textual representation of the
pass-pipeline to run. It is expected that the textual representation is
for a pass-manager already nested on a func-like op.

---------

Signed-off-by: MaheshRavishankar <mahesh.ravishankar@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

nirvedhmeshram committed:
[GPU] Pad fusion support for TileAndDistributeToWorkgroupsUsingForall (#20258)

Do not generate zero slice guard by using an explicit pattern to swap
extract slice with pad in TileAndDistributeToWorkgroupsUsingForall.
Fixes :  https://github.com/iree-org/iree/issues/20253

Signed-off-by: Nirvedh Meshram <nirvedh@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

bjacob committed:
Make `math.fpowi` rewrite more systematic and test it. (#20293)

This is a follow-up to #20281 which lacked a test, and had left the c++
logic in a state that was prone to unintentional regression. Also, this
ensures that the `math.fpowi` rewrite happens even with
`-iree-codegen-gpu-native-math-precision`, which is intentional as
`math.fpowi` is not really a math function, like the comment explains.

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

kuhar committed:
[Codegen] Disallow padding more skinny matmuls (#20289)

Fix up the threshold from https://github.com/iree-org/iree/pull/20284 to
disallow other skinny matmuls. Add a TODO to refactor this in the
future.

Also add debug prints for padding with
`--debug-only=iree-encoding-attrs`.

---------

Signed-off-by: Jakub Kuderski <jakub@nod-labs.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Groverkss committed:
[GPU] Improve Memory Bound Attention Heuristics (#20280)

The previous constraints were really naive. This patch adds constraints
that assume a specific attention KV layout and tries to optimize for
them. Generally, we shouldn't have any other layout and if we do, we
should realistically change the layout at the model level. The layout is
described in inline comments in the code, but TLDR is that K1/N should
be kept as inner dimensions and K2 should be kept as an outer dimension.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

IanWood1 committed:
[Global Opt] Prevent expanding reduction dims (#20290)

Prevents fusing reshapes with reduction ops. Fixes llama fp8 perf
regression due to interleaved parallel/reduction dimensions after llvm
integrate
(https://github.com/iree-org/llvm-project/commit/813bbe055df2357efd42eaf1f3e7f5926a071941).

Adding `memref::populateResolveRankedShapedTypeResultDimsPatterns` is
unrelated, but I noticed some dynamic dimensions that weren't getting
simplified during this pass.

---------

Signed-off-by: Ian Wood <ianwood2024@u.northwestern.edu>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

hanhanW committed:
[Encoding] Implement isIdentityLayout interface method. (#20286)

The revision introduces `isIdentityLayout` interface method to
SerializableEncodingAttrInterface, and implements the method for
`EncodingAttr` and `PadEncodingLayoutAttr` attributes.

It also fixes a bug in `EncodingAttr` builders, where we should use NULL
when maps are not present.

---------

Signed-off-by: hanhanW <hanhan0912@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

IanWood1 committed:
[docs] Mention TRACY_PORT in tracy profiling docs (#20296)

This updates the tracy profiling documentation to  add a small mention of `TRACY_PORT`
below the section that discusses changing the port of the `iree-tracy-capture`.

---------

Signed-off-by: Ian Wood <ianwood2024@u.northwestern.edu>
Co-authored-by: Han-Chung Wang <hanhan0912@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved regression test on to dedicated runner

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved regression test on to dedicated runner

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved regresstion test back because benchmark server has no drivers installed

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
moved regresstion test back because benchmark server has no drivers installed

Signed-off-by: Elias Joseph <eljoseph@amd.com>

bjacob committed:
Deprecate `--iree-codegen-gpu-native-math-precision` (#20299)

This was mostly used on ROCm and WebGPU.

On ROCm, over the past month we have made this be essentially the
default behavior, so the flag is essentially not needed anymore. The
last thing required before we can drop usage of the flag on ROCm is
https://github.com/iree-org/iree/pull/20074.

On WebGPU, this PR makes this be the default behavior like on ROCm, so
this PR also drops the flag. The WGSL spec issue
https://github.com/gpuweb/gpuweb/issues/5109 mentioned in the comment
explains the problem. It has also been discussed in the past as
https://github.com/iree-org/iree/issues/11321.

Signed-off-by: Benoit Jacob <jacob.benoit.1@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

hanhanW committed:
[Stream] Implement folders for stream.tensor.encode op. (#20298)

There are few folders available for `stream.tensor.encode` op, we can
fold it into the source resource for the cases:
- The source encoding is as the same as the result encoding.
- If they are not identical, fold the op when the source layout and the
result layout are identity layouts.

A tensor type without encodings is recognized as an identity layout. A
tensor type with the encoding is an identity layout if the encoding
attribute implements the `isIdentityLayout` interface method and it
returns true.

---------

Signed-off-by: hanhanW <hanhan0912@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

qedawkins committed:
Integrate llvm-project@681b2413 (#20274)

Retains reverts for 418e07b7e and 1a3dc8e9c.

---------

Signed-off-by: MaheshRavishankar <mahesh.ravishankar@gmail.com>
Co-authored-by: MaheshRavishankar <mahesh.ravishankar@gmail.com>
Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd commented on .github/workflows/pkgci_test_sharktank.yml:
```diff
@@ -62,13 +67,13 @@ jobs:
         run: |
           source ${VENV_DIR}/bin/activate
-          python -m pip install -r iree-test-suites/sharktank_models/requirements.txt
+          python -m pip install -r iree-test-suites/sharktank_models/requirements-iree.txt
```
See https://github.com/iree-org/iree/pull/20302 which also updates the iree-test-suites version used here. We should get your patch(es) landed on `main` there and do a regular update, not pin to a branch.

From that PR, this should also be `python -m pip install iree-test-suites/sharktank_models`. The requirements-iree pulls in _nightly_ IREE packages, when what this test in iree-org/iree should be doing is installing _dev_ IREE packages.

Eliasj42 replied:
is iree test suit updated, because last week the tests there didnt work

ScottTodd replied:
See the PR I linked.

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
```diff
@@ -95,6 +95,14 @@ jobs:
         if: "!cancelled()"
         run: |
           source ${VENV_DIR}/bin/activate
+          # Set IREE_TEST_FILES based on matrix.name
+          case "${{ matrix.name }}" in
+            "amdgpu_rocm_mi300_gfx942") IREE_TEST_FILES="/shark-cache/data/iree-regression-cache" ;;
+            *) echo "No cache directory assigned for ${{ matrix.name }}" ;;
+          esac
+          if [[ -n "$IREE_TEST_FILES" ]]; then
+            export IREE_TEST_FILES
+          fi
```
Thanks for adding the `echo` here, that would help with understanding what the workflow is doing. I do have two alternate suggestions that I think will be more natural to work with though:

1. Set the cache environment variable on the runners themselves. This allows different runners to set different cache locations, and we needed that before when we had multiple mi300 runners, each using different cache locations but sharing the same runner labels. The downsides of that are that it is lower visibility in the workflows themselves and it is more work on the runner configuration side, especially when setting up runners for many different github projects.
2. Set the cache environment variable in the matrix. I _think_ this is possible, going off of https://stackoverflow.com/questions/75446952/github-workflows-changing-or-specifying-environment-variables-in-matrix-jobs and https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow. Not sure about the exact syntax that would work in this case though.

    So rather than have
    ```yml
          matrix:
            include:
              - name: amdgpu_rocm_mi250_gfx90a
                runs-on: nodai-amdgpu-mi250-x86-64
              - name: amdgpu_rocm_mi300_gfx942
                runs-on: linux-mi300-1gpu-ossci-iree-org
    
    ...
          - name: Set cache dir
            run: |
              # Set IREE_TEST_FILES based on matrix.name
              case "${{ matrix.name }}" in
                "amdgpu_rocm_mi300_gfx942") echo "IREE_TEST_FILES=/shark-cache/data/iree-regression-cache" >> $GITHUB_ENV
                *) echo "No cache directory assigned for ${{ matrix.name }}" ;;
              fi
    ```

    We would have something more like

    ```yml
          matrix:
            include:
              - name: amdgpu_rocm_mi250_gfx90a
                runs-on: nodai-amdgpu-mi250-x86-64
              - name: amdgpu_rocm_mi300_gfx942
                runs-on: linux-mi300-1gpu-ossci-iree-org
                env:
                  IREE_TEST_FILES: /shark-cache/data/iree-regression-cache
    ```

Given that each runner label is currently consistent in how the cache is set up, I think option 2 is a good fit here. Could you give that a try?

Eliasj42 replied:
I tried that and the environmental variable wasn't used

Eliasj42 replied:
https://github.com/iree-org/iree/commit/3a1fdbc33795a22ef735f86c5b46eca9848de166

ScottTodd replied:
Check with @renxida , he's doing something similar right now on https://github.com/nod-ai/shark-ai/pull/1108 too

ScottTodd commented on .github/workflows/pkgci_test_sharktank.yml:
```diff
@@ -30,7 +30,12 @@ jobs:
+          - name: rocm_hip_mi300
+            target: target_hip
+            gpu: gfx942
+            runs-on: linux-mi300-2gpu-ossci-iree-org
```
Is this running the 2 GPU tests or do we also need to opt-in somehow? When logs are ready, can you link to them here showing which tests are running?

It's also fine to keep the scope of this PR small and just add a new 1 GPU test, or even fully separate the `test_sharktank.yml` changes from the `regression_test.yml` changes.

Eliasj42 replied:
I'm pretty sure it is running the 2 gpu test, but Ill link it when the logs finish

qedawkins committed:
[Codegen] Fix mma distribution tests to use valid input (#20303)

The upstream commit 418e07b7 added static slice verification that
revealed a few tests with invalid inputs. This updates said tests and
drops the upstream revert.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

benvanik committed:
Emit an error message if a device alias is used but is missing targets. (#20288)

Fixes #20283.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

qedawkins committed:
[Codegen] Replace upstream constant bufferization pass with our own (#20304)

Recently the upstream constructor for the OneShotBufferizePass was
refactored and no longer supported some of the options we needed. Our
use case for it was fairly specific so this replaces it with our own
IREEBufferizeConstantsPass.

This also drops the revert getting us to a clean state with integrate.

Signed-off-by: Elias Joseph <eljoseph@amd.com>

krzysz00 committed:
[Codegen][ROCDL] Use buffer fat pointers where possible (#19918)

This commit enables using the `#amdgpu.address_space<fat_raw_buffer>`
address space for memref subspans in our kernels in cases where it's
safe to do so. This both gains us immediate performance improvements
(-2ms on unet) and enables further performance improvements on MI-300
through the use of the cache swizzling mechanism.

Note: this PR currently depends on a bunch of open upstream work.

# Context

AMD GPUs, on top of having normal `global_{load,store,...}_T`
instructions, which take a 64-bit pointer, also have `buffer_*`
instructions, which use the texture unit to enable more complex logic.

These instructions, for our purposes, take a "buffer resource" (a base
pointer into global memory + flags) and an offset. The resource takes 4
**scalar** registers (and thus must be wave-uniform) and the offset is 1
**vector** register, which is added to the base pointer to form the
effective address of the buffer access.

This means that we can use these buffer operations if we can be assured
that the non-uniform part of our accesses never exceeds 2 GB from the
start of whatever we decide the base pointer is.

IREE will often pass in N copies of the same pointer and then take
global offsets from that pointer to locate the logical bindings/tensors
within the model's data. Fortunately, these offsets are globally
uniform.

Therefore, we can conservatively construct buffer descriptors by finding
the start of a given bound tensor/memref (which is always in a
contiguous layout) and using that as a base pointer. Then, if we know
that the memref is 2 GB or smaller, we can use the buffer instructions
to access it.

# How this patch works

1. Before bufferization, we run iree-rocdl-configure-buffer-instructions
/ ROCDLConfigureBufferInstructions , which will annotate
`hal.interface.binding.subspan` operations with the key `
iree_gpu.use_rocdl_buffer_instructions` as a discardable attribute on
the subspans *if* we can prove that said subspans are safe to use as
buffer instructions
2. During bufferization, if ` iree_gpu.use_rocdl_buffer_instructions` is
set, then we use the `#amdgpu.address_space<buffer_fat_pointer>` memory
space instead of the `#hal.descriptor_type<*>` one. This required
refactoring a few passes that were using `#hal.descriptor_type` as a
synonym for "global memory" in tests, but that was a fairly small change
3. Finally, in
`iree-codegen-convert-hal-descriptor-type-to-gpu-address-space`, we
implement the buffer fat pointer ABI by converting these buffer bindings
to normal global bindings and then adding an
`amdgpu.fat_raw_buffer_cast` with the `resetOffset` option set to fold
in the global byte offset. (I'm not entirely happy with this part and
maybe we should move it to bufferization)

So it's recorded: +~3% on SDXL on MI-300 from this

Signed-off-by: Elias Joseph <eljoseph@amd.com>

rsuderman committed:
Bump iree-org/iree-test-suites@56509666d185cdb99aafb4ed4d8532b8f3c54fa3 (#20302)

Signed-off-by: Rob Suderman <rob.suderman@gmail.com>

Commit:
updated ref

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
updated ref

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Eliasj42 committed:
Merge branch 'main' into users/eliasj42/move-pkgci-to-ossci-cluster

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
removed bugged pip insatll

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
more sanity check

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed style issues

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
fixed style issues

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
reverted changes because they didnt work

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
reverted changes because they didnt work

Commit:
cahgned back

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
cahgned back

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
tested to see if syntax error

Signed-off-by: Elias Joseph <eljoseph@amd.com>

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
```diff
@@ -95,6 +97,7 @@ jobs:
         if: "!cancelled()"
         run: |
           source ${VENV_DIR}/bin/activate
+          [ -z "$IREE_TEST_FILES" ] && unset IREE_TEST_FILES
```
What are these `unset` commands trying to do? They caused the CPU tests to redownload files and timeout/fail: https://github.com/iree-org/iree/actions/runs/13979151202/job/39140779239?pr=20260#step:6:142

Eliasj42 replied:
If I don't unset IREE_TEST_FILES, the when I try to move the envionmental variable over it sets _REE_TEST_FILES to ''

ScottTodd commented on .github/workflows/pkgci_test_sharktank.yml:
```diff
@@ -69,6 +74,7 @@ jobs:
         run: |
           source ${VENV_DIR}/bin/activate
           python -m pip install iree-test-suites/sharktank_models/
+          python -m pip install -r iree-test-suites/sharktank_models/requirements-iree.txt
```
Remove this. The workflow already installed dev IREE packages (testing dev packages on pull request and push events is the whole point of these workflows), but this command installs nightly packages: https://github.com/iree-org/iree-test-suites/blob/main/sharktank_models/requirements-iree.txt

Eliasj42 replied:
ok, but it was seg faulting without this

ScottTodd replied:
Focus on one thing at a time. This line doesn't belong here. I'd suggest creating one PR for each workflow instead.

ScottTodd reviewed (changes requested):
Is this actually ready for review? There are multiple checks failing on the PR:

* new sharktank_models on mi300: https://github.com/iree-org/iree/actions/runs/13979151202/job/39140779295?pr=20260 (cancelled)
* regression suite on CPU: https://github.com/iree-org/iree/actions/runs/13979151202/job/39140779239?pr=20260
* DCO: https://github.com/iree-org/iree/pull/20260/checks?check_run_id=39140272215

Commit:
addressed comments

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
addressed comments

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
added conditional env setting

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
added conditional env setting

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
hopefully this goes to the right branch this time

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
hopefully this goes to the right branch this time

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
maybe this actually does something

Signed-off-by: Elias Joseph <eljoseph@amd.com>

Commit:
Merge branch 'users/eliasj42/move-pkgci-to-ossci-cluster' of https://github.com/iree-org/iree into users/eliasj42/move-pkgci-to-ossci-cluster
